src/main/java/org/mastodon/util/DummySpimData.java

```diff
@@ -139,22 +144,65 @@ private static double get( final String[] parts, final String key, final double
 		return defaultValue;
 	}
 
-	static class DummyImgLoader implements BasicImgLoader
+	static class DummyImgLoader implements ViewerImgLoader
 	{
-		private final BasicSetupImgLoader< ? > setupImgLoader;
+		private final ViewerSetupImgLoader< ?, ? > setupImgLoader;
 
 		public DummyImgLoader()
 		{
 			this( new UnsignedShortType(), new FinalDimensions( 100, 100, 100 ) );
 		}
 
-		public < T > DummyImgLoader( final T type, final Dimensions dimensions )
+		public < T extends RealType< T >> DummyImgLoader(final T type, final Dimensions dimensions )
 		{
 			assert ( dimensions.numDimensions() == 3 );
+			final FinalInterval dimInterval = new FinalInterval( dimensions );
 
-			final RandomAccessibleInterval< T > img = Views.interval( new ConstantRandomAccessible<>( type, 3 ), new FinalInterval( dimensions ) );
-			setupImgLoader = new BasicSetupImgLoader< T >()
+			final RandomAccessibleInterval< T > img = Views.interval( new ConstantRandomAccessible<>( type, 3 ), dimInterval );
+			final RandomAccessibleInterval< Volatile< T > > volatileImg = Views.interval( new ConstantRandomAccessible<>( new VolatileRealType<>( type ), 3 ), dimInterval );
+
+			setupImgLoader = new ViewerSetupImgLoader< T, Volatile< T > >()
 			{
+				@Override
+				public RandomAccessibleInterval<T> getImage(int timepointId, int level, ImgLoaderHint... hints) {
+					return img;
+				}
+
+				final double[][] resolution = new double[1][3];
+				final AffineTransform3D[] transform = new AffineTransform3D[1];
+				{
+					resolution[0][0] = 1.0; //TODO should be taken from 'calib' from our only-one caller code...
```

Comment:
I'm not sure how much important is to fix this TODO given that we talk here about DUMMY data...

if we were to make it crystal clear, we would need to modify the c'tor (easy, there is only one caller of it) to bring in ["calibration" and "transformation"](https://github.com/mastodon-sc/mastodon/blob/adc7bddb4850510007b1a93004f33d04b7bb2340/src/main/java/org/mastodon/util/DummySpimData.java#L110-L120)